cli/azd/pkg/tools/azcli/deployments.go

```diff
@@ -109,7 +111,8 @@ func (cli *azCli) DeployToSubscription(
 	// wait for deployment creation
 	deployResult, err := createFromTemplateOperation.PollUntilDone(ctx, nil)
 	if err != nil {
-		return AzCliDeploymentResult{}, fmt.Errorf("deploying to subscription: %w", err)
+		deploymentError := createDeploymentError(err)
+		return AzCliDeploymentResult{}, fmt.Errorf("deploying to subscription: %w", deploymentError)
```

Comment:
Do we want to include:

```suggestion
		return AzCliDeploymentResult{}, fmt.Errorf("deploying to subscription:\n\nDeployment Error Details:\n%w", deploymentError)
```